### PR TITLE
EnqueuePlugin added

### PR DESCRIPTION
**Pull request by sanyokerem**

### CODE REVIEW

**Comment by sanyokerem** [comment]
The phpunit had to be updated because the previous version is not supporting type declaration for return values. 
https://github.com/sebastianbergmann/phpunit/issues/1785

File: .gitignore [hunk 1]
```diff
@@ -3,3 +3,4 @@
 composer.lock
 phpunit.xml
 vendor/
+/.idea
```
Comment (cmodijk) [inline .gitignore]: This is a editor specific ignore and we generally don't add these in open source package

File: src/Plugin/EnqueuePlugin/ObjectManager.php [hunk 2]
```diff
@@ -0,0 +1,9 @@
+<?php
+
+namespace LongRunning\Plugin\EnqueuePlugin;
+
+
+interface ObjectManager extends \Doctrine\Common\Persistence\ObjectManager
+{
+    public function isOpen();
+}
```
Comment (cmodijk) [inline src/Plugin/EnqueuePlugin/ObjectManager.php]: Could you add a line ending to this file?

File: src/Plugin/EnqueuePlugin/WhenPostMessageReceivedCleanUp.php [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+    public function __construct(Cleaner $cleaner)
+    {
+        $this->cleaner = $cleaner;
+    }
+
+    public function onPostMessageReceived(PostMessageReceived $context) : void
+    {
+        $this->cleaner->cleanUp();
+    }
+}
```
Comment (cmodijk) [inline src/Plugin/EnqueuePlugin/WhenPostMessageReceivedCleanUp.php]: Could you add a line ending to this file?

File: tests/Plugin/EnqueuePlugin/WhenPostMessageReceivedCleanUpTest.php [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+            $this->createMock(Consumer::class),
+            $this->createMock(Message::class),
+            null,
+            1,
+            $this->createMock(LoggerInterface::class)
+        );
+
+        return $messageReceived;
+    }
+}
```
Comment (cmodijk) [inline tests/Plugin/EnqueuePlugin/WhenPostMessageReceivedCleanUpTest.php]: Could you add a line ending to this file?

**Comment by cmodijk** [comment]
The 5.6 and 7.0 versions are not supported i created a new issue for this to drop those #27

**Comment by makasim** [comment]
@cmodijk the php5.6 and 7 have been dropped. tests pass.

**Review by cmodijk — CHANGES_REQUESTED** [review]
Can you also look at the comments on the code?

File: src/Plugin/EnqueuePlugin/ObjectManager.php [hunk 5]
```diff
@@ -0,0 +1,9 @@
+<?php
+
+namespace LongRunning\Plugin\EnqueuePlugin;
```
Comment (makasim) [inline src/Plugin/EnqueuePlugin/ObjectManager.php]: I think it should be moved to Tests namespace

Comment (cmodijk) [reply src/Plugin/EnqueuePlugin/ObjectManager.php]: Yes i agree

File: src/Plugin/EnqueuePlugin/ObjectManager.php (hunk 2 above)
Comment (cafferata) [reply src/Plugin/EnqueuePlugin/ObjectManager.php]: I also see a unnecessary double blank line in this file.

**Comment by cmodijk** [comment]
I'm not sure if we should drop 5.6 in this same pull request @ruudk what do you think? Merge this and then also add php 7.2?

**Review by cmodijk — APPROVED** [review]

File: .travis.yml [hunk 6]
```diff
@@ -4,8 +4,6 @@ language: php
 
 matrix:
   include:
-    - php: 5.6
-    - php: 7.0
```
Comment (ruudk) [inline .travis.yml:8]: @sanyokerem Can you remove this commit, and rebase your branch with master? Then we can merge it.

Comment (makasim) [reply .travis.yml:8]: Enqueue requires php7.1+ so could not be rebased with master.

Comment (ruudk) [reply .travis.yml:8]: Ok, wait for https://github.com/LongRunning/LongRunning/pull/30 to be merged

Comment (cmodijk) [reply .travis.yml:8]: #30 is merged can you rebase this, we can then merge it and release a new version.

**Comment by ruudk** [comment]
@sanyokerem Can you rebase this PR with master so that we can merge it?

**Pull request closed without merging**